frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts

```diff
@@ -0,0 +1,106 @@
+    nav.sidenav.switcher.changePerspectiveTo('Administrator');
+    nav.sidenav.switcher.shouldHaveText('Administrator');
+    cy.createProject(testName);
+  });
+
+  beforeEach(() => {
+    nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
+    cy.byLegacyTestID('resource-title').should('have.text', 'Secrets');
+    listPage.clickCreateKeyValSecretDropdownButton();
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  after(() => {
+    cy.deleteProject(testName);
+    cy.logout();
+  });
+
+  it(`Adds a key/value secret whose value is a binary file `, () => {
+    cy.get('.co-m-pane__heading').contains('Create key/value secret');
+    cy.byTestID('secret-name').should('exist');
+    cy.byLegacyTestID('file-input-textarea').should('exist');
+    cy.byTestID('secret-name').type(binarySecretName);
+    cy.byTestID('secret-key').type(secretKey);
+    cy.byTestID('file-input').attachFile('binarysecret.bin');
+    cy.byLegacyTestID('file-input-textarea').should('not.exist');
+    cy.get(infoMessage).should('exist');
+    cy.byTestID('save-changes').click();
+    detailsPage.isLoaded();
+    cy.exec(
+      `oc get secret -n ${testName} ${binarySecretName} --template '{{.data.${secretKey}}}' | base64 -d`,
+      {
+        failOnNonZeroExit: false,
+      },
+    ).then((value) => {
+      cy.fixture('binarysecret.bin', 'binary').then((binarySecret) => {
+        expect(binarySecret).toEqual(value.stdout);
+      });
+    });
+  });
+
+  it(`Adds a key/value secret whose value is an ascii file `, () => {
+    cy.get('.co-m-pane__heading').contains('Create key/value secret');
+    cy.byTestID('secret-name').should('exist');
+    cy.byLegacyTestID('file-input-textarea').should('exist');
+    cy.byTestID('secret-name').type(asciiSecretName);
+    cy.byTestID('secret-key').type(secretKey);
+    cy.byTestID('file-input').attachFile('asciisecret.txt');
+    cy.byLegacyTestID('file-input-textarea').should('exist');
+    cy.get(infoMessage).should('not.exist');
+    cy.byTestID('save-changes').click();
+    detailsPage.isLoaded();
+    cy.exec(
+      `oc get secret -n ${testName} ${asciiSecretName} --template '{{.data.${secretKey}}}' | base64 -d`,
+      {
+        failOnNonZeroExit: false,
+      },
+    ).then((value) => {
+      cy.fixture('asciisecret.txt', 'ascii').then((asciiSecret) => {
+        expect(asciiSecret.trim()).toEqual(value.stdout);
+      });
+    });
+  });
+
+  it(`Adds a key/value secret whose value is a unicode file `, () => {
+    cy.get('.co-m-pane__heading').contains('Create key/value secret');
+    cy.byTestID('secret-name').should('exist');
+    cy.byLegacyTestID('file-input-textarea').should('exist');
+    cy.byTestID('secret-name').type(unicodeSecretName);
+    cy.byTestID('secret-key').type(secretKey);
+    cy.byTestID('file-input').attachFile('unicodesecret.utf8');
+    cy.byLegacyTestID('file-input-textarea').should('exist');
+    cy.get(infoMessage).should('not.exist');
+    cy.byTestID('save-changes').click();
```

Comment:
Since this is repeated code, consider making a function `attachFile(secretName, secretKey, file)`